networkx/algorithms/tree/branchings.py

```diff
@@ -384,6 +397,16 @@ def _init(self, attr, default, kind, style, preserve_attrs, seed, partition):
         # being different).
         self.minedge_circuit = []
 
+    """
+    We should seperate each step into an inner function. Then the overall loop would become
+    while True:
+        step_I1()
+        if cycle detected:
+            step_I2()
+        elif every node of G is in D and E is a branching
+            break
+    """
+
```

Comment:
Generally speaking, it's best to only use strings to comment code in the docstrings. For body comments, use literal comments. If you use strings like this, the interpreter actually allocates a string object that bloats the memory footprint even though it's not used for anything!

```suggestion
    # We should seperate each step into an inner function. Then the overall loop would become
    # while True:
    #     step_I1()
    #     if cycle detected:
    #         step_I2()
    #     elif every node of G is in D and E is a branching
    #         break

```

Reply:
That was written by whoever wrote the original Edmonds class. Since that code will eventually be deprecated, I decided not to remove it.